modules/offers/client/components/Offers.component.js

```diff
@@ -0,0 +1,374 @@
+      default:
+        return 'Cannot host currently';
+    }
+  }
+
+  setOfferDescriptionToggle(state) {
+    this.setState(() => ({
+      offerDescriptionToggle: state // !prevState.offerDescriptionToggle
+    })
+    );
+  }
+
+  renderButtonOwn() {
+    const { offer, hostingStatusLabel } = this.state;
+    return (
+      <div
+        className="pull-right btn-group"
+        uib-dropdown
+        is-open="hostingDropdown">
+        <button
+          uib-tooltip="Change"
+          tooltip-placement="left"
+          uib-dropdown-toggle
+          className={classnames(
+            'btn', 'btn-sm', 'dropdown-toggle', 'btn-offer-hosting',
+            { 'btn-offer-hosting-yes': offer.status === 'yes',
+              'btn-offer-hosting-maybe': offer.status === 'maybe',
+              'btn-offer-hosting-no': (!offer || offer.status === 'no')
+            })}>
+          { hostingStatusLabel(offer.status) }
+          <span className="caret"></span>
+        </button>
+        <ul className="dropdown-menu" role="menu">
+          <li>
+            <a ui-sref="offer.host.edit({'status': 'yes'})"
+              className="cursor-pointer offer-hosting-yes">
+              I can host
+            </a>
+          </li>
+          <li>
+            <a ui-sref="offer.host.edit({'status': 'maybe'})"
+              className="cursor-pointer offer-hosting-maybe">
+              I might be able to host
+            </a>
+          </li>
+          <li>
+            <a ui-sref="offer.host.edit({'status': 'no'})"
+              className="cursor-pointer offer-hosting-no">
+              {'I can\'t host currently'}
+            </a>
+          </li>
+        </ul>
+      </div>
+    );
+  }
+
+  renderButtonOther() {
+    const { offer, hostingStatusLabel } = this.state;
+    return (
+      <a aria-label="Hosting status: {{ hostingStatusLabel(offer.status) }}"
+        ui-sref="messageThread({username: profile.username})"
+        className={classnames(
+          'btn', 'btn-sm', 'pull-right', 'btn-offer-hosting', 'btn-offer-hosting-yes',
+          {
+            'btn-offer-hosting-no': !offer || offer.status === 'no',
+            'btn-offer-hosting-yes': offer.status === 'yes',
+            'btn-offer-hosting-maybe': offer.status === 'maybe'
+          })}>
+        { hostingStatusLabel(offer.status) }
+      </a>
+    );
+  }
+
+  renderHostingYesMaybe() {
+    const { offer, offerDescriptionToggle, isOwnOffer } = this.state;
+    return (
+      <div>
+        {/*  Edit button  */}
+        {isOwnOffer &&
+          <a ui-sref="offer.host.edit"
+            className="btn btn-inverse-primary btn-round btn-raised pull-right"
+            aria-label="Modify hosting offer">
+            <span className="icon-edit"></span>
+          </a>
+        }
+
+        {/*  Short descriptions  */}
+        {(offer.description && offer.description.length < 2000) &&
+          <div dangerouslySetInnerHTML={{ __html: sanitizeHtml(offer.description) }}>
```

Comment:
This will break description texts since default sanitization opens are different from APIs.